classes/notifier/models/reminder/course_non_participation_model.php

```diff
@@ -49,17 +49,17 @@ public function get_user_ids_to_notify() {
 
         global $DB;
 
-        $results = $DB->get_records_sql('SELECT u.id
+        $results = $DB->get_records_sql("SELECT u.id
             FROM {user} u
             INNER JOIN {user_enrolments} ue ON ue.userid = u.id
             INNER JOIN {enrol} e ON e.id = ue.enrolid
             INNER JOIN {course} c ON c.id = e.courseid
             INNER JOIN {role_assignments} ra ON ra.userid = u.id
             INNER JOIN {context} ctx ON ctx.id = ra.contextid AND ctx.instanceid = c.id
             WHERE u.id NOT IN (SELECT la.userid FROM {user_lastaccess} la WHERE la.courseid = c.id AND la.timeaccess > ?)
-            AND ra.roleid IN (SELECT value FROM {config} WHERE name = "gradebookroles")
+            AND ra.roleid IN (SELECT cast(value as int) FROM {config} WHERE name = 'gradebookroles')
```

Comment:
does this method `cast()` work in both mysql and psql databases? I'm reading some psql examples that it should be `CAST (value as INTEGER)` on psql whereas its `CAST(value as int)` on mysql?

Reply:
my bad, I didn't notice the difference... and yeah, I am reading some docs, they write `cast value as Integer` for psql.